src/main/java/br/codenation/hospital/manager/model/Hospital.java

```diff
@@ -51,5 +56,53 @@ public Hospital(
     this.availableBeds = availableBeds;
     this.latitude = latitude;
     this.longitude = longitude;
+    this.patients = new HashMap<>();
+    this.stock = new HashMap<>();
+  }
+
+  public Patient addNewPatient(Patient patient) {
+
+    if (!this.patients.containsKey(patient.getId())) {
+      this.patients.put(patient.getId(), patient);
+      return patient;
+    }else
+      return null;
+  }
+
+  public Patient removePatient(String patientId){
+    return this.patients.remove(patientId);
+  }
+
+  public SupplyItem addItemStock(SupplyItem supplyItem){
```

Comment:
Another suggestion

```java
if (patients.containsKey(supplyItem.getId())) {
      final long currentQuantity = supplyItem.getQuantity();
      supplyItem.setQuantity(currentQuantity + stock.get(supplyItem.getId()).getQuantity());

      return stock.replace(supplyItem.getId(), supplyItem);
    }

    return stock.putIfAbsent(supplyItem.getId(), supplyItem);
```

Reply:
You mean this?  stock.containsKey instead patients.containsKey?

`if (stock.containsKey(supplyItem.getId())) {
      final long currentQuantity = supplyItem.getQuantity();
      supplyItem.setQuantity(currentQuantity + stock.get(supplyItem.getId()).getQuantity());

      return stock.replace(supplyItem.getId(), supplyItem);
    }

    return stock.putIfAbsent(supplyItem.getId(), supplyItem);`

Reply:
yes, my bad